### DISCUSSION TITLE
How to change starship color configuration based on light or dark mode

### DISCUSSION THREAD

**Discussion by tkolleh** [starship/starship#4821] [Q&A]
Is there a way to change or reload starship configurations from a script?

The following function is added to `~/.zshrc` (see [dot files](https://github.com/tkolleh/dotfiles/blob/master/dot_zshrc.tmpl#L29)). It changes Starship prompt **configuration file reference** based on Apple appearance settings (light / dark mode) and the active iTerm2 profile. My dev experience is such that all CLI tools change colors when the OS switches to dark mode at dusk. Issue is that Starship does not switch not even after `source ~/.zshrc`. I am forced to start a new session. 

Whats a more seamless approach to switching configurations?

Configuration in `~/.zshrc` (see [dot files](https://github.com/tkolleh/dotfiles/blob/master/dot_zshrc.tmpl#L29)):

```zsh
set_prompt_mode() {
  # Change Starship prompt configuration based on Apple appearance 
  # setting (light / dark mode) and the active iTerm2 profile
  if is_dark_mode && [[ $ITERM_PROFILE == "Github" ]]; then
      echo "Starship going dark"
      export STARSHIP_CONFIG="$HOME/.config/starship_dark.toml"
  fi
}
```

**Comment by walid-mos** [comment]
Hello, this is quite old but i managed to do it by using dark mode notify script by bouk :
https://github.com/bouk/dark-mode-notify

Need to make install (you will need xcode terminal), create a daemon as in readme, which will launch a script going from one theme to another 

Here is my script : 
```
STARSHIP_CONFIG="/Users/YOURUSER/.config/starship/starship.toml"

get_theme() {
    if [ "$DARKMODE" = "1" ]; then
        echo "catppuccin_mocha"
    else
        echo "catppuccin_latte"
    fi
}

update_starship_config() {
    local theme=$(get_theme)
    sed -i '' "s/^palette = .*/palette = \"$theme\"/" "$STARSHIP_CONFIG"
    echo "Configuration Starship mise à jour" >&2
    echo "$(date): Thème détecté - $theme" >> "$log_file"
    echo "$(date): Configuration mise à jour" >> "$log_file"
}

update_starship_config

```
